pkg/helpers/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,115 @@
+package imageregistry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/imageregistry/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func AddToScheme(s *runtime.Scheme) error {
+	if err := clusterv1.Install(s); err != nil {
+		return err
+	}
+	if err := v1alpha1.AddToScheme(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+type ImageRegistryClient struct {
+	client client.Client
+}
+
+func NewImageRegistry(client client.Client) *ImageRegistryClient {
```

Comment:
it will not be easy if the consumer is not using  controller-runtime

Reply:
referring your suggestions, I define a Client interface and create 2 objects, one uses controller-time, another uses kubeconfig.